pyface/gui_application.py

```diff
@@ -167,6 +172,8 @@ def start(self):
             # create the GUI so that the splash screen comes up first thing
             if self.gui is Undefined:
                 self.gui = GUI(splash_screen=self.splash_screen)
+                if self.splash_screen_duration > 0:
+                    sleep(self.splash_screen_duration)
```

Comment:
This halts all processing for a number of seconds, and in particular the event loop will not be running so the app will be locked-up, which is generally not a good thing.  At a minimum, this should occasionally be processing events.